### PR TITLE
Upgraded to sgx 2.0, updated code with safe funcs

### PR DESCRIPTION
**Pull request by askmish**

### CODE REVIEW

**Comment by dcmiddle** [comment]
@askmish FYI, looks like a build problem. You can see the errors in the console logs through Jenkins (follow the details link under "Some checks were not successful"
https://build.sawtooth.me/job/Sawtooth-Hyperledger/job/sawtooth-core/job/PR-1196/6/console

File: consensus/poet/common/c11_support/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-attributes")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+    SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -m64")
+    SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -O2")
+    SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
+ENDIF()
+
+#ADD_DEFINITIONS(-D LIBPOET_BUILD)
```
Comment (cianx) [inline consensus/poet/common/c11_support/CMakeLists.txt]: Remove these unused lines.

File: sdk/c/loader.c [hunk 2]
```diff
@@ -24,6 +24,8 @@
 #include <openssl/engine.h>
 #include <openssl/conf.h>
 
+#define strncpy_s strncpy
```
Comment (cianx) [inline sdk/c/loader.c]: If this passes a scanning tool... we need a better scanning tool. There are still occurrences of `strncpy` in this file, those should be updated.

**Comment by jsmitchell** [comment]
Hi @askmish - we typically use a rebase workflow to keep the commit history clean of merge commits. Could you rewrite your commits to remove that 2625fa0 merge commit?

File: consensus/poet/common/c11_support/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+CMAKE_MINIMUM_REQUIRED(VERSION 3.2 FATAL_ERROR)
+PROJECT(c11_support C CXX)
```
Comment (cianx) [inline consensus/poet/common/c11_support/CMakeLists.txt]: Is C support required?

File: consensus/poet/common/c11_support/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+ENDIF()
+
+#ADD_DEFINITIONS(-D LIBPOET_BUILD)
+#SET_RUNTIME(DYNAMIC)
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+# Compile targets
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+# CMake likes to prepend "lib" to project name when creating library...since we
```
Comment (cianx) [inline consensus/poet/common/c11_support/CMakeLists.txt]: This comment is wrong and the removal of the prefix is probably not necessary for this case.

**Comment by askmish** [comment]
Need sgx sdk 2.0 for building sgx code, else ci builds will fail at the latest sgx defs.

**Comment by cianx** [comment]
@TomBarnes do you have any issue with @vaporos proposal to delay this commit to post 1.0?

**Comment by TomBarnes** [comment]
This change is required to satisfy Sawtooth SDL requirements for release 1.0.  It provides the safe function library required to remove banned C++ functions from the PoET enclave code.

**Comment by vaporos** [comment]
After the docs are updated to reflect SGX 2.0, we will test this PR on the LR0 hardware.  Specifically, I think we are missing an update to the "Install SGX/PSW" section.

**Comment by askmish** [comment]
@vaporos The docs have been updated including the "Install SGX/PSW" section, as part of the commits.

**Comment by vaporos** [comment]
@askmish Great! Sorry I missed that when I scanned through.  We will do some testing tomorrow.

**Pull request merged**